[RFC] vim-patch:8.0.0131 and vim-patch:8.0.0155

ckelsel opened the pull request:
vim-patch:8.0.0131
    
    Problem:    Not enough test coverage for syntax commands.
    Solution:   Add more tests. (Dominique Pelle)
    
    https://github.com/vim/vim/commit/73b484c4da00011317dc68ada4f5dfc6515ad263
    
    vim-patch 8.0.0155
    
    Problem:    When sorting zero elements a NULL pointer is passed to qsort(),
                which ubsan warns for.
    Solution:   Don't call qsort() if there are no elements. (Dominique Pelle)
    
    https://github.com/vim/vim/commit/a216255a4faa91a15e7005ac319f2f62294f3f9e

rprichard committed:
win: integrate winpty (WIP)

Handling of process exit is still broken.  It detects the moment when the
child process exits, then quickly stops polling for process output.  It
should continue polling for output until the agent has scraped all of the
process' output.  This problem is easy to notice by running a command like
"dir && exit", but even typing "exit<ENTER>" can manifest the problem --
the "t" might not appear.

winpty's Cygwin adapter handles shutdown by waiting for the agent to close
the CONOUT pipe, which it does after it has scraped the child's last
output.  AFAIK, neovim doesn't do anything interesting when winpty closes
the CONOUT pipe.

Commit:
win/install: winpty-agent.exe

justinmk committed:
ui: disable 'lazyredraw' during ui_refresh.

Could also try `do_redraw = true` instead of save/restore `p_lz`, but
the nice thing about save/restore of `p_lz` is that it is "atomic".
The semantics of `do_redraw` are not clear to me.

Closes #4884
References #6202
References https://github.com/neovim/neovim/pull/6202#issuecomment-284379503
References #3929 #5692 #6157
References #5866

erw7 committed:
win: support :terminal

erw7 committed:
win/pty: jobstart, jobstop: fix null-pointer dereference

- Make sure that proc->in is not NULL, because nvim crashed when
  starting a job with pty.
- Make sure that proc->out is not NULL, because nvim crashed when stopping
  a job opened with pty.

erw7 committed:
win/pty: quote_cmd_arg(): check bounds

erw7 committed:
win/test: tty-test: print screen size explicitly with CTRL-Q

tty-test.exe causes abnormal termination with low repeatability, try
changing it so as not to use SIGWINCH.

erw7 committed:
win/pyt: cleanup

erw7 committed:
win/pty: log errors

erw7 committed:
pty_process_win: avoid quoting for cmd.exe

nelstrom committed:
Use Normal mode as default when opening a new terminal

nelstrom committed:
Repair ex_terminal_spec functional tests

nelstrom committed:
Repair job_spec functional tests

nelstrom committed:
Repair tui_spec functional tests

nelstrom committed:
Update documentation

justinmk committed:
vim-patch:8.0.0678 closing a window does not trigger resizing

Closes #6748

Problem:    When 'equalalways' is set and closing a window in a separate
            frame, not all window sizes are adjusted. (Glacambre)
Solution:   Resize all windows if the new current window is not in the same
            frame as the closed window. (closes vim/vim#1707)

https://github.com/vim/vim/commit/8eeeba8c025ff844e6514c4a60cec11bf1fc1b35

ZyX-I committed:
functests: Comment out failing test

ZyX-I committed:
functests: Make tests work with input()

There are still some issues: specifically, new “pending” test hangs busted.

justinmk committed:
vim-patch:8.0.0679

Problem:    Using freed memory.
Solution:   Get the parent frame pointer earlier.

https://github.com/vim/vim/commit/41cc038ff83498c589c7d25b3d2984145528eb92

ZyX-I committed:
functests: Fix typo

ZyX-I committed:
functests: Make “stops executing callback” test work

Needed to be adjusted to use input() (previously relied on side-effects of 
executing `:cmd`) and dismiss something (hidden “Press ENTER” message?).

ZyX-I committed:
ex_getln: Call highlight callback inside :try

ZyX-I committed:
functests: Partially uncomment <C-c> test

ZyX-I committed:
functests: Uncomment `{REDRAW}` part of “works” test

ZyX-I committed:
ex_getln: Fix “echoerr msg not shown” problem

This also attempted to fix problem with cancelling input() on error by avoiding 
standard error printing facilities (assumed thrown error message is the 
problem), but with no luck so far.

ZyX-I committed:
ex_getln: Make sure standard error reporting facility is not used

ZyX-I committed:
ex_getln: Fix indent

ZyX-I committed:
ex_getln: Do not make <C-c> interrupt input() after interrupting hl cb

ckelsel committed:
fix hostname_spec.lua test failed

ckelsel committed:
test

ckelsel committed:
rm test

ZyX-I committed:
ex_getln: Save and restore try state

Problem: when processing cycle such as

    :for pat in [' \ze*', ' \zs*']
    :  try
    :    let l = matchlist('x x', pat)
    :    $put ='E888 NOT detected for ' . pat
    :  catch
    :    $put ='E888 detected for ' . pat
    :  endtry
    :endfor

`:let l = …` throwing an error causes this error to be caught after 
color_cmdline attempts to get callback for highlighting next line (the one with 
`$put = 'E888 NOT…`). Saving/restoring state prevents this from happening.

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

DarkDefender committed:
tui: fix DECSCUSR logic #6997

Fix linuxvt cursor shape codes
Fix konsole cursor_shapes (even when inside tmux)
Do not trust old VTE terminal lies

Closes #6978
Closes #7002
Closes #7049

ckelsel committed:
ignore patch-2367,2364

ZyX-I committed:
Merge branch 'master' into colored-cmdline

ckelsel committed:
merge upstream/master

ZyX-I committed:
oldtests: Use $(TOOL) in place of $(VALGRIND)

ZyX-I committed:
api helpers: Save/restore more values in try_enter/try_leave

This fixes memory leak reported by ASAN. This also somehow fixes test40, though 
I have no idea why except that that test yields memory leak report.

ZyX-I committed:
ex_getln: Do not goto color_cmdline_end without first cleaning up

The issue with debug mode was actually not cleaning up after `try_enter`: 
location `&tstate` was pointing to got invalidated and received some “garbage” 
(actually, values that got stored on the stack afterwards). But pointer to that 
garbage was still stored in `msg_list`, so next attempt to check it resulted in 
a crash.

ZyX-I committed:
ex_getln: Do not do useless try_enter/try_leave calls

These are actually needed for two modes only. And even for these modes they 
should eventually go away.

ZyX-I committed:
doc: Adjust documentation a bit

2 spaces after sentense ends, clarify what will be highlighted.

ZyX-I committed:
ex_getln: Replace global with entry in save_ccline

ZyX-I committed:
api helpers: Also save and restore did_emsg

ZyX-I committed:
ex_getln: Check prev_prompt_errors before running redrawcmdline

Otherwise there will be infinite recursion and shortly a crash. Running 
redrawcmdline recursively occurs under color_cmdline_error label.

ZyX-I committed:
functests: Check for previously unchecked errors

ZyX-I committed:
functests: Test input() nesting support

ZyX-I committed:
*: Fix clint errors

ZyX-I committed:
ex_getln: Avoid GCC “unused variable” warning from QB

    17:25:45,363 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/ex_getln.c: In function ‘color_cmdline’:
    17:25:45,363 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/ex_getln.c:2335:8: error: variable ‘printed_errmsg’ set but not used [-Werror=unused-but-set-variable]
    17:25:45,363 WARN  - bool printed_errmsg = false;
    17:25:45,363 WARN  - ^
    17:25:45,399 WARN  - cc1: all warnings being treated as errors

ZyX-I committed:
functests: Add sleep to <C-c> test

ckelsel committed:
vim-patch:8.0.0044

Problem:    In diff mode the cursor may end up below the last line, resulting
            in an ml_get error.
Solution:   Check the line to be valid.

https://github.com/vim/vim/commit/025e3e0bafbc85cc4e365145af711edf99d0a90d

ckelsel committed:
merge upstream

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ZyX-I committed:
doc: Update documentation

ZyX-I committed:
functests: Remove unneeded wait()s

ZyX-I committed:
message: Only require second (format) argument to be not NULL

ZyX-I committed:
functests: Replace wait() with nvim_async

ZyX-I committed:
ex_getln: Cache highlight callback calling results

justinmk committed:
log: some DEBUG-level stream logging

justinmk committed:
rpc: close channel if stream was closed

f_jobstop()/f_rpcstop() .. process_stop() .. process_close_in(proc)
closes the write-stream of a RPC channel. But there might be
a pending RPC notification on the queue, which may get processed just
before the channel is closed.

To handle that case, check the Stream.closed in
channel.c:receive_msgpack().

Before this change, the above scenario could trigger
this assert(!stream->closed) in wstream_write():

    0x00007f96e1cd3428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
    0x00007f96e1cd502a in __GI_abort () at abort.c:89
    0x00007f96e1ccbbd7 in __assert_fail_base (fmt=<optimized out>, assertion=assertion@entry=0x768f9b "!stream->closed",
    file=file@entry=0x768f70 "../src/nvim/event/wstream.c", line=line@entry=77,
    function=function@entry=0x768fb0 <__PRETTY_FUNCTION__.13735> "wstream_write") at assert.c:92
    0x00007f96e1ccbc82 in __GI___assert_fail (assertion=0x768f9b "!stream->closed", file=0x768f70 "../src/nvim/event/wstream.c", line=77,
    function=0x768fb0 <__PRETTY_FUNCTION__.13735> "wstream_write") at assert.c:101
    0x00000000004d2c1f in wstream_write (stream=0x7f96e0a35078, buffer=0x7f96e09f9b40) at ../src/nvim/event/wstream.c:77
    0x00000000005857b2 in channel_write (channel=0x7f96e0ae5800, buffer=0x7f96e09f9b40) at ../src/nvim/msgpack_rpc/channel.c:551
    0x000000000058567d in on_request_event (argv=0x7ffed792efa0) at ../src/nvim/msgpack_rpc/channel.c:523
    0x00000000005854c8 in handle_request (channel=0x7f96e0ae5800, request=0x7ffed792f1b8) at ../src/nvim/msgpack_rpc/channel.c:503
    0x00000000005850cb in parse_msgpack (channel=0x7f96e0ae5800) at ../src/nvim/msgpack_rpc/channel.c:423
    0x0000000000584f90 in receive_msgpack (stream=0x7f96e0a35218, rbuf=0x7f96e0d1d4c0, c=22, data=0x7f96e0ae5800, eof=false)
    at ../src/nvim/msgpack_rpc/channel.c:389
    0x00000000004d0b20 in read_event (argv=0x7ffed792f4a8) at ../src/nvim/event/rstream.c:190
    0x00000000004ce462 in multiqueue_process_events (this=0x7f96e18172d0) at ../src/nvim/event/multiqueue.c:150
    0x000000000059b630 in nv_event (cap=0x7ffed792f620) at ../src/nvim/normal.c:7908
    0x000000000058be69 in normal_execute (state=0x7ffed792f580, key=-25341) at ../src/nvim/normal.c:1137
    0x0000000000652463 in state_enter (s=0x7ffed792f580) at ../src/nvim/state.c:61
    0x000000000058a1fe in normal_enter (cmdwin=false, noexmode=false) at ../src/nvim/normal.c:467
    0x00000000005500c2 in main (argc=2, argv=0x7ffed792f8d8) at ../src/nvim/main.c:554

Alternative approach suggested by bfredl is to use close_cb of the
process. My unsuccessful attempt is below. (It seems close_cb is queued
too late, which is the similar problem addressed by this commit):

    commit 75fc12c6ab15711bdb7b18c6d42ec9d157f5145e
    Author: Justin M. Keyes <justinkz@gmail.com>
    Date:   Fri Aug 18 01:30:41 2017 +0200

        rpc: use Stream's close_cb instead of explicit check in receive_msgpack()

    diff --git a/src/nvim/event/process.c b/src/nvim/event/process.c
    index 8371d3cd482e..e52da23cdc40 100644
    --- a/src/nvim/event/process.c
    +++ b/src/nvim/event/process.c
    @@ -416,6 +416,10 @@ static void on_process_exit(Process *proc)
     static void on_process_stream_close(Stream *stream, void *data)
     {
       Process *proc = data;
    +  ILOG("on_process_stream_close");
    +  if (proc->stream_close_cb != NULL) {
    +    proc->stream_close_cb(stream, proc->stream_close_data);
    +  }
       decref(proc);
     }

    diff --git a/src/nvim/event/process.h b/src/nvim/event/process.h
    index 5c00e8e7ecd5..34a8d54f6f8c 100644
    --- a/src/nvim/event/process.h
    +++ b/src/nvim/event/process.h
    @@ -26,6 +26,11 @@ struct process {
       Stream *in, *out, *err;
       process_exit_cb cb;
       internal_process_cb internal_exit_cb, internal_close_cb;
    +
    +  // Called when any of the process streams (in/out/err) closes.
    +  stream_close_cb stream_close_cb;
    +  void *stream_close_data;
    +
       bool closed, detach;
       MultiQueue *events;
     };
    @@ -50,6 +55,8 @@ static inline Process process_init(Loop *loop, ProcessType type, void *data)
         .closed = false,
         .internal_close_cb = NULL,
         .internal_exit_cb = NULL,
    +    .stream_close_cb = NULL,
    +    .stream_close_data = NULL,
         .detach = false
       };
     }
    diff --git a/src/nvim/event/stream.c b/src/nvim/event/stream.c
    index 7c865bfe1e8c..c8720d1e45d9 100644
    --- a/src/nvim/event/stream.c
    +++ b/src/nvim/event/stream.c
    @@ -95,7 +95,11 @@ void stream_close(Stream *stream, stream_close_cb on_stream_close, void *data)
     void stream_close_handle(Stream *stream)
       FUNC_ATTR_NONNULL_ALL
     {
    +  ILOG("stream=%d", stream);
    +  // LOG_CALLSTACK();
       if (stream->uvstream) {
    +    // problem:  this schedules on the queue, but channel.c:receive_msgpack may
    +    // be processed before close_cb is called by libuv.
         uv_close((uv_handle_t *)stream->uvstream, close_cb);
       } else {
         uv_close((uv_handle_t *)&stream->uv.idle, close_cb);
    @@ -105,6 +109,7 @@ void stream_close_handle(Stream *stream)
     static void close_cb(uv_handle_t *handle)
     {
       Stream *stream = handle->data;
    +  ILOG(">>>>>>>>>>>>>>>>>>>>>>> stream=%p stream->internal_close_cb=%p", stream, stream->internal_close_cb);
       if (stream->buffer) {
         rbuffer_free(stream->buffer);
       }
    diff --git a/src/nvim/msgpack_rpc/channel.c b/src/nvim/msgpack_rpc/channel.c
    index 782eabe04e4a..dc2b794e366a 100644
    --- a/src/nvim/msgpack_rpc/channel.c
    +++ b/src/nvim/msgpack_rpc/channel.c
    @@ -128,6 +128,8 @@ uint64_t channel_from_process(Process *proc, uint64_t id, char *source)
                                           source);
       incref(channel);  // process channels are only closed by the exit_cb
       channel->data.proc = proc;
    +  channel->data.proc->stream_close_cb = close_cb2;
    +  channel->data.proc->stream_close_data = channel;

       wstream_init(proc->in, 0);
       rstream_init(proc->out, 0);
    @@ -387,17 +389,6 @@ static void receive_msgpack(Stream *stream, RBuffer *rbuf, size_t c,
         goto end;
       }

    -  if ((chan_wstream(channel) != NULL && chan_wstream(channel)->closed)
    -      || (chan_rstream(channel) != NULL && chan_rstream(channel)->closed)) {
    -    char buf[256];
    -    snprintf(buf, sizeof(buf),
    -             "ch %" PRIu64 ": stream closed unexpectedly. "
    -             "closing channel",
    -             channel->id);
    -    call_set_error(channel, buf, WARN_LOG_LEVEL);
    -    goto end;
    -  }
    -
       size_t count = rbuffer_size(rbuf);
       DLOG("ch %" PRIu64 ": parsing %u bytes from msgpack Stream: %p",
            channel->id, count, stream);
    @@ -571,23 +562,6 @@ static Stream *chan_wstream(Channel *chan)
       abort();
     }

    -/// Returns the Stream that a Channel reads from.
    -static Stream *chan_rstream(Channel *chan)
    -{
    -  switch (chan->type) {
    -    case kChannelTypeSocket:
    -      return &chan->data.stream;
    -    case kChannelTypeProc:
    -      return chan->data.proc->out;
    -    case kChannelTypeStdio:
    -      return &chan->data.std.in;
    -    case kChannelTypeInternal:
    -      return NULL;
    -  }
    -  abort();
    -}
    -
    -
     static bool channel_write(Channel *channel, WBuffer *buffer)
     {
       bool success = false;
    @@ -799,6 +773,12 @@ static void close_cb(Stream *stream, void *data)
       decref(data);
     }

    +static void close_cb2(Stream *stream, void *data)
    +{
    +  ILOG("close_cb2");
    +  close_channel(data);
    +}
    +
     /// @param source description of source function, rplugin name, TCP addr, etc
     static Channel *register_channel(ChannelType type, uint64_t id,
                                      MultiQueue *events, char *source)

ZyX-I committed:
Revert "functests: Replace wait() with nvim_async"

This reverts commit e129607988b88719935bc4af517e7ee2689f5871.

Tests stopped working in CI.

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

justinmk committed:
vim-patch:8.0.0518

Closes #7086

Problem:    Storing a zero byte from a multi-byte character causes fold text
            to show up wrong.
Solution:   Avoid putting zero in ScreenLines. (Christian Brabandt,
            closes vim/vim#1567)

https://github.com/vim/vim/commit/c6cd8409c2993b1476e123fba11cb4b8d743b896

justinmk committed:
vim-patch:8.0.0311

Problem:    Linebreak tests are old style.
Solution:   Turn the tests into new style. Share utility functions. (Ozaki
            Kiichi, closes vim/vim#1444)

https://github.com/vim/vim/commit/544d3bc9f0e494cb712a33b61558b8e8e12b1e0b

justinmk committed:
vim-patch:8.0.0090 fix breakindent bug

(original Vim commit-message is bogus)

https://github.com/vim/vim/commit/6c896867c4f5d759616028ef7cbfce2a9ed32600

justinmk committed:
vim-patch:8.0.0290 vim-patch:8.0.0394

vim-patch:8.0.0290: cursor positioning wrong if wide character wraps
Problem:    If a wide character doesn't fit at the end of the screen line, and
            the line doesn't fit on the screen, then the cursor position may
            be wrong. (anliting)
Solution:   Don't skip over wide character. (Christian Brabandt, closes vim/1408)

vim-patch:8.0.0394
Problem:    Tabs are not aligned when scrolling horizontally and a Tab doesn't
            fit. (Axel Bender)
Solution:   Handle a Tab as a not fitting character. (Christian Brabandt)
            Also fix that ":redraw" does not scroll horizontally to show the
            cursor.  And fix the test that depended on the old behavior.

https://github.com/vim/vim/commit/abc39ab642791ae3d22a524516eeedb673a95d9d

justinmk committed:
vim-patch:8.0.0126

Problem:    Display problem with 'foldcolumn' and a wide character.
            (esiegerman)
Solution:   Don't use "extra" but an allocated buffer. (Christian Brabandt,
            closes vim/vim#1310)

https://github.com/vim/vim/commit/6270660611a151c5d0f614a5f0248ccdc80ed971

justinmk committed:
vim-patch:8.0.0128

Problem:    Display test fails on MS-Windows.
Solution:   Set 'isprint' to "@".

https://github.com/vim/vim/commit/7089237885218eb8a19805bc2b75481c4efcd6ba

justinmk committed:
vim-patch:8.0.0524

Problem:    Folds are messed up when 'encodin' is "utf-8".
Solution:   Also set the fold character when it's not multi-byte.

https://github.com/vim/vim/commit/8da1e6cedf839902e15987a98733ebd31b5f1b81

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
vim-patch:8.0.0131

Problem:    Not enough test coverage for syntax commands.
Solution:   Add more tests. (Dominique Pelle)

https://github.com/vim/vim/commit/73b484c4da00011317dc68ada4f5dfc6515ad263

ckelsel committed:
vim-patch:8.0.0131

Problem:    Not enough test coverage for syntax commands.
Solution:   Add more tests. (Dominique Pelle)

https://github.com/vim/vim/commit/73b484c4da00011317dc68ada4f5dfc6515ad263

vim-patch:8.0.0132
Problem:    Test fails because of using :finish.
Solution:   Change to return.

https://github.com/vim/vim/commit/4c8980b717f73042f1d625ee255fa74eddb989ba

justinmk commented:
```
/home/travis/build/neovim/neovim/src/nvim/syntax.c:5849:9: runtime error: null pointer passed as argument 1, which is declared to never be null
/usr/include/stdlib.h:766:30: note: nonnull attribute specified here
    #0 0x17d599b in syntime_report /home/travis/build/neovim/neovim/src/nvim/syntax.c:5849:3
    #1 0x17d3400 in ex_syntime /home/travis/build/neovim/neovim/src/nvim/syntax.c:5762:5
    #2 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #3 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #4 0xb74be5 in do_cmdline_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:279:10
    #5 0x90c1a8 in f_execute /home/travis/build/neovim/neovim/src/nvim/eval.c:8015:5
    #6 0x817cc3 in call_func /home/travis/build/neovim/neovim/src/nvim/eval.c:6408:11
    #7 0x82d5c3 in get_func_tv /home/travis/build/neovim/neovim/src/nvim/eval.c:6148:11
    #8 0x8b9b16 in eval7 /home/travis/build/neovim/neovim/src/nvim/eval.c:4248:15
    #9 0x8b603b in eval6 /home/travis/build/neovim/neovim/src/nvim/eval.c:3945:7
    #10 0x8b3947 in eval5 /home/travis/build/neovim/neovim/src/nvim/eval.c:3796:7
    #11 0x8aec4f in eval4 /home/travis/build/neovim/neovim/src/nvim/eval.c:3533:7
    #12 0x8ae12c in eval3 /home/travis/build/neovim/neovim/src/nvim/eval.c:3451:7
    #13 0x8ad5fc in eval2 /home/travis/build/neovim/neovim/src/nvim/eval.c:3382:7
    #14 0x8101e2 in eval1 /home/travis/build/neovim/neovim/src/nvim/eval.c:3309:7
    #15 0x80f5fa in eval0 /home/travis/build/neovim/neovim/src/nvim/eval.c:3269:9
    #16 0x81c0db in ex_let /home/travis/build/neovim/neovim/src/nvim/eval.c:1499:9
    #17 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #18 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #19 0x84d87d in call_user_func /home/travis/build/neovim/neovim/src/nvim/eval.c:21337:3
    #20 0x8175e7 in call_func /home/travis/build/neovim/neovim/src/nvim/eval.c:6393:11
    #21 0x82d5c3 in get_func_tv /home/travis/build/neovim/neovim/src/nvim/eval.c:6148:11
    #22 0x825d3b in ex_call /home/travis/build/neovim/neovim/src/nvim/eval.c:2760:9
    #23 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #24 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #25 0x870405 in ex_execute /home/travis/build/neovim/neovim/src/nvim/eval.c:19520:7
    #26 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #27 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #28 0x84d87d in call_user_func /home/travis/build/neovim/neovim/src/nvim/eval.c:21337:3
    #29 0x8175e7 in call_func /home/travis/build/neovim/neovim/src/nvim/eval.c:6393:11
    #30 0x82d5c3 in get_func_tv /home/travis/build/neovim/neovim/src/nvim/eval.c:6148:11
    #31 0x825d3b in ex_call /home/travis/build/neovim/neovim/src/nvim/eval.c:2760:9
    #32 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #33 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #34 0xb5b140 in do_source /home/travis/build/neovim/neovim/src/nvim/ex_cmds2.c:2935:3
    #35 0xb57604 in cmd_source /home/travis/build/neovim/neovim/src/nvim/ex_cmds2.c:2680:14
    #36 0xb57730 in ex_source /home/travis/build/neovim/neovim/src/nvim/ex_cmds2.c:2661:3
    #37 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #38 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #39 0xb74be5 in do_cmdline_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:279:10
    #40 0xeb8ecb in exe_commands /home/travis/build/neovim/neovim/src/nvim/main.c:1644:5
    #41 0xea6ba2 in main /home/travis/build/neovim/neovim/src/nvim/main.c:506:5
    #42 0x2ae6b0f1ff44 in __libc_start_main /build/eglibc-SvCtMH/eglibc-2.19/csu/libc-start.c:287
    #43 0x44d22b in _start (/home/travis/build/neovim/neovim/build/bin/nvim+0x44d22b)

SUMMARY: AddressSanitizer: undefined-behavior /home/travis/build/neovim/neovim/src/nvim/syntax.c:5849:9 in 
```

ZyX-I committed:
Merge branch 'master' into colored-cmdline

ZyX-I committed:
Merge branch 'master' into colored-cmdline

ckelsel commented:
@justinmk I don't know how to fix this. It's a little difficult for me.

ckelsel committed:
Merge branch 'vim-8.0.0131' of github.com:ckelsel/neovim into vim-8.0.0131

Conflicts:
	src/nvim/testdir/test_syntax.vim
	src/nvim/version.c

ckelsel committed:
Merge remote-tracking branch 'upstream/master' into vim-8.0.0131

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

justinmk commented:
No problem. I or someone else will look at it.

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
Merge remote-tracking branch 'upstream/master' into vim-8.0.0131

Conflicts:
	src/nvim/version.c

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

justinmk committed:
nvim -h: omit special-case options

Group some options, and sort them alphabetically.
`nvim -h` should fit on one (smallish) screen.
Uncommon options don't need to be here, they live in the :help.

ZyX-I committed:
api/helpers: Clarify try_start() usage

ZyX-I committed:
functests: Remove wait() from input_spec

ZyX-I committed:
api/helpers: Fix typo

ZyX-I committed:
Merge branch 'master' into colored-cmdline

justinmk committed:
test: cleanup

justinmk committed:
single-includes: ignore os/pty_process_win.h

justinmk committed:
test: tty-test.c: restore win32 SIGWINCH handler

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

justinmk committed:
win/test: enable more :terminal tests

To deal with SIGWINCH limitations on Windows, change some resize tests
to _shrink_ the screen width. ... But this didn't work, so still
ignoring those tests on Windows.

oni-link commented:
Test failure because of missing vim-patch:8.0.0155.

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

ckelsel committed:
vim-patch:8.0.0174

Problem:    For completion "locale -a" is executed on MS-Windows, even though
            it most likely won't work.
Solution:   Skip executing "locale -a" on MS-Windows. (Ken Takata)

https://github.com/vim/vim/commit/b8f7bd68f6fdff51ca4db5b145dcde42ee7b804b

justinmk committed:
oldtest: cannot `:set term` in Nvim

ckelsel committed:
Merge remote-tracking branch 'upstream/master'

justinmk committed:
vim-patch:8.0.0235

Problem:    Memory leak detected when running tests for diff mode.
Solution:   Free p_extra_free.

https://github.com/vim/vim/commit/b031c4ea04eb1e37a873fbb85e90d835aa1e2b1c

justinmk committed:
test: tty-test.c: keep `tty_out` handle around

Now the window_split_tab_spec.lua test seems to work.
Also do some cleanup.

jamessan committed:
travis: Move TSAN to last stage and allow failure

TSAN build has been much less reliable lately, so it shouldn't hold up
the other tests.

ZyX-I committed:
lua/executor: Fix crash when printing empty string (#7157)

justinmk committed:
io: more guards against NULL filename (#7159)

References ac055d677aa9
References #4370

justinmk committed:
test/win: place cursor at edge to tickle SIGWINCH

ZyX-I committed:
functests: Check that input is correctly silenced

ZyX-I committed:
doc: Clarify how function is executed

ZyX-I committed:
functests: Alter comment

ZyX-I committed:
doc: State that it is called for new *displayed* input

ZyX-I committed:
ex_getln: Make error messages look better

justinmk committed:
doc: screen_resize(): remove mention of "mustset"

"mustset" is from Vim term.c:set_shellsize(), not relevant here: we
behave as if mustset=true always.

ckelsel committed:
Merge remote-tracking branch 'upstream/master' into HEAD

ckelsel committed:
vim-patch:8.0.0131

Problem:    Not enough test coverage for syntax commands.
Solution:   Add more tests. (Dominique Pelle)

https://github.com/vim/vim/commit/73b484c4da00011317dc68ada4f5dfc6515ad263

jamessan committed:
ci: Switch to codecov.io for coverage data

Continue uploading to coveralls, for now, so we can compare the
services.

ckelsel committed:
Merge branch 'master' into vim-8.0.0131

ckelsel committed:
Merge branch 'master' into vim-8.0.0131

ckelsel committed:
Merge branch 'vim-8.0.0131' of github.com:ckelsel/neovim into vim-8.0.0131

ckelsel committed:
vim-patch:8.0.0131

Problem:    Not enough test coverage for syntax commands.
Solution:   Add more tests. (Dominique Pelle)

https://github.com/vim/vim/commit/73b484c4da00011317dc68ada4f5dfc6515ad263

vim-patch 8.0.0155

Problem:    When sorting zero elements a NULL pointer is passed to qsort(),
            which ubsan warns for.
Solution:   Don't call qsort() if there are no elements. (Dominique Pelle)
https://github.com/vim/vim/commit/a216255a4faa91a15e7005ac319f2f62294f3f9e

ckelsel committed:
vim-patch:8.0.0131

Problem:    Not enough test coverage for syntax commands.
Solution:   Add more tests. (Dominique Pelle)

https://github.com/vim/vim/commit/73b484c4da00011317dc68ada4f5dfc6515ad263

vim-patch 8.0.0155

Problem:    When sorting zero elements a NULL pointer is passed to qsort(),
            which ubsan warns for.
Solution:   Don't call qsort() if there are no elements. (Dominique Pelle)

https://github.com/vim/vim/commit/a216255a4faa91a15e7005ac319f2f62294f3f9e

oni-link commented on src/nvim/syntax.c:
```diff
@@ -5845,9 +5845,12 @@ static void syntime_report(void)
-  qsort(ga.ga_data, (size_t)ga.ga_len, sizeof(time_entry_T),
-      syn_compare_syntime);
+  // Sort on total time. Skip if there are no items to avoid passing NULL
+  // pointer to qsort().
```
Skip ... -> Skip if there is nothing to sort.

justinmk committed:
Merge #6259 from justinmk/ui_refresh

jamessan committed:
ci: Collect coverage data for MINGW_64 builds on appveyor

jamessan committed:
ci: Ignore QB for codecov's CI status

jamessan committed:
Merge pull request #7161 from jamessan/codecov

ci: Upload coverage data to codecov.io

Closes #7162

justinmk committed:
ci/win: list build permutations explicitly (#7163)

This avoids changing the matrix permutations for the old non-gcov
builds, so that old URLs continue to work:

https://ci.appveyor.com/api/projects/neovim/neovim/artifacts/build/Neovim.zip?branch=master&job=Configuration%3A%20MINGW_32
https://ci.appveyor.com/api/projects/neovim/neovim/artifacts/build/Neovim.zip?branch=master&job=Configuration%3A%20MINGW_64

teto committed:
syntax.c: register 'Normal' highlight group

- :hi Normal works with -u NONE
- Makes HL_TABLE and ATTR_ENTYRY a function instead of a macro so that in can be used in gdb.
- Introduces ATTRENTRY_INIT to init attrentry_t

jamessan committed:
ci: Rename .codecov.yml → codecov.yml

According to codecov/support#431 the comment setting isn't honored when
the yaml file is named .codecov.yml.

teto committed:
syntax.c: style

Converts some documentation to doxygen format + minor styling
improvements.

teto committed:
coverty: solve  #7149

jamessan committed:
Merge pull request #7167 from jamessan/codecov

ci: Rename .codecov.yml → codecov.yml

justinmk committed:
Merge #6364 'command-line color hook'

justinmk committed:
doc

justinmk committed:
test/win: give up on this one

jamessan committed:
Merge pull request #7168 from teto/fix_coverty

Closes #7149

jamessan committed:
ci: Ignore MINGW_64-gcov failures in overall ci status

jamessan committed:
Merge pull request #7174 from jamessan/appveyor-allow-cov-failure

ci: Ignore MINGW_64-gcov failures in overall ci status

jamessan committed:
Merge pull request #6928 from justinmk/vim-patch-8.0.0678

vim-patch:8.0.0678 closing a window does not trigger resizing

jamessan committed:
Merge pull request #7145 from ckelsel/vim-8.0.0174

vim-patch:8.0.0174

jamessan committed:
Merge pull request #7052 from ckelsel/vim-8.0.0044

vim-patch:8.0.0044

justinmk committed:
Merge #7007 'Windows :terminal'

References #6383
Closes #4748

justinmk committed:
intro: remove byline #6984

justinmk committed:
version: tweak layout, doc

justinmk committed:
doc/api: nvim_out_write() and friends

References #7178

justinmk committed:
channel.c:call_set_error(): fix memory leak

jamessan committed:
provider#clear_stderr: Use remove() not delete() to update s:stderr

Ref #7184

jamessan committed:
Merge pull request #7185 from jamessan/fix-provider-clear_stderr

provider#clear_stderr: Use remove() not delete() to update s:stderr

justinmk committed:
Merge #7171 from justinmk/doc

justinmk committed:
clipboard: avoid error flood during :redir

redir_write():
- This is a "batch" operation which was not yet covered by
  start_batch_changes()

adjust_clipboard_name():
- msg() and friends during :redir will, of course, cause redir_write()
  to try to capture that message, which causes recursion.
- EMSG() here is trouble: if it interrupts :redir it is a mess.
  Rather than deal with the mess, show a non-error message.

closes #7182
closes #7184
closes #7183
ref #6048
ref #7032

llorens committed:
vim-patch:8.0.0092

Problem:    C indenting does not support nested namespaces that C++ 17 has.
Solution:   Add check that passes double colon inside a name. (Pauli, closes
            vim/vim#1214)

https://github.com/vim/vim/commit/ca8b8d6956dd881de6446fc32c38e817a364a6cc

justinmk committed:
clipboard: remove start_batch_changes() in redir_write()

start_batch_changes() doesn't avoid invoking the clipboard
once-per-line, because the loop is actually in ex_echo(), which calls
redir_write() for each message. But we've already entered
start_batch_changes() by then, so that was never the problem.

    redir_write at /home/vagrant/old.neovim/build/../src/nvim/message.c:2523
    msg_puts_attr_len at /home/vagrant/old.neovim/build/../src/nvim/message.c:1600
    msg_outtrans_len_attr at /home/vagrant/old.neovim/build/../src/nvim/message.c:1221
    ex_echo at /home/vagrant/old.neovim/build/../src/nvim/eval.c:19433
    do_one_cmd at /home/vagrant/old.neovim/build/../src/nvim/ex_docmd.c:2242

Trying to defer _explicit_ clipboard updates is difficult.
    :redir @+ | silent echo system('cat foo') | redir END
is essentially equivalent to:
    for l in readfile('foo')
        let @+ .= l
    endfor
We cannot make judgements about when to ignore a script's bad decisions.
start_batch_changes() only works around the case of clipboard=unnamed,
i.e. _implicit_ clipboard updates (`:g/foo/d`).  Not explicit
assignment.

justinmk committed:
clipboard: test g:clipboard validation, fix a bug

Also fix `:help foo` highlighting in health.vim

justinmk committed:
Merge #7193 from justinmk/cb-pathology

jamessan committed:
vim-patch:8.0.0308

Problem:    When using a symbolic link, the package path will not be inserted
            at the right position in 'runtimepath'. (Dugan Chen, Norio Takagi)
Solution:   Resolve symbolic links when finding the right position in
            'runtimepath'. (Hirohito Higashi)

https://github.com/vim/vim/commit/2f9e575583c2ad3978ee3d0f790eeff7df56bd6c

jamessan committed:
vim-patch:8.0.0325

Problem:    Packadd test does not clean up symlink.
Solution:   Delete the link. (Hirohito Higashi)

https://github.com/vim/vim/commit/913727e56761d57aaba61197c2d3485418dea7eb

jamessan committed:
vim-patch:8.0.0326

Problem:    Packadd test uses wrong directory name.
Solution:   Use the variable name value. (Hirohito Higashi)

https://github.com/vim/vim/commit/24f8f543d4036c5d2ce4ea6973a174cf2176cb72

jamessan committed:
vim-patch:8.0.0437

Problem:    The packadd test does not create the symlink correctly and does
            not test the right thing.
Solution:   Create the directory and symlink correctly.

https://github.com/vim/vim/commit/644df41c44cbdfacdedbba55ef77a6c6031eccd8

jamessan committed:
vim-patch:8.0.0612

Problem:    Package directories are added to 'runtimepath' only after loading
            non-package plugins.
Solution:   Split off the code to add package directories to 'runtimepath'.
            (Ingo Karkat, closes vim/vim#1680)

https://github.com/vim/vim/commit/ce876aaa9a250a5a0d0e34b3a2625e51cf9bf5bb

jamessan committed:
vim-patch:8.0.0680

Problem:    Plugins in start packages are sourced twice. (mseplowitz)
Solution:   Use the unmodified runtime path when loading plugins (test by Ingo
            Karkat, closes vim/vim#1801)

https://github.com/vim/vim/commit/07ecfa64a18609a986f21d6132d04ee8934f3200

jamessan committed:
Merge pull request #6808 from nelstrom/normal-mode-terminal

Make :terminal remain in normal mode when created

fornwall committed:
tui: always use unibi_add_ext_str with unibi_get_ext_str #7204

When using an index returned by unibi_add_ext_str() we should
always use unibi_get_ext_str() and not rely on the index being
lower than unibi_string_begin_.

Closes #7206

jamessan committed:
Merge pull request #7201 from jamessan/vim-8.0.0308

vim-patch:8.0.0308,8.0.0325,8.0.0326,8.0.0437,8.0.0612,8.0.0680

justinmk committed:
Merge #6973 from teto/normal_hl

justinmk committed:
clipboard: disallow recursion; show hint only once (#7203)

- Show hint only once per session.
- provider#clipboard#Call(): prevent recursion
- provider#clear_stderr(): use has_key(), because :silent! is still
  captured by :redir.

closes #7184

yuntan committed:
win: wmain(): locale-independent argv (#7180)

fix #7060

justinmk committed:
win: wmain(): use utf16_to_utf8() #7060

justinmk committed:
win: expect utf8-encoded `argv` when built as a library

justinmk committed:
Merge #7205 from justinmk/win-wmain

lbschenkel committed:
tui: support new iterm2, iTerm2.app terminfo entries

iTerm2 got its own entry in Thomas Dickey's terminfo.src on 2017-08-16.
Make sure that the new entry is handled in the same way as the old entry.

closes #7209
closes #7214

jamessan committed:
Merge pull request #7192 from llorens/vim-8.0.0092

vim-patch:8.0.0092

justinmk committed:
Merge #7081 from justinmk/rpcstop

rpc: close channel if stream was closed

Commit:
ci: XXX: permissive termclose_spec timer (#7217)

We probably need to revisit the process_stop() logic.
In the meantime this hack avoids CI noise.

jamessan committed:
provider: Remove dict attribute from stderr_collector

If an autoloaded function hasn't been resolved before it is used in
function(), the self dict will not be created which causes E725 when
calling the function.  Since self isn't being used in
provider#stderr_collector, we can remove the dict attribute to
workaround the self dict bug[0].

Closes #7115

[0]: https://groups.google.com/d/msg/vim_dev/I7AXOyv-P4o/DzbyOxDHBgAJ

jamessan committed:
Merge pull request #7218 from jamessan/stderr_collector-fix

provider: Remove dict attribute from stderr_collector

gpanders committed:
health.vim: always check pyenv if installed #7219

Always check for the presence of pyenv_root if pyenv is installed: if it
is not set, we don't know if it was intentional. If it wasn't
intentional, the warning is confusing (see #7176).

closes #7176

justinmk committed:
eventloop: FocusGained: schedule event instead of pseudokey

closes #4840
closes #6164

KillTheMule committed:
inccommand: fix optimization logic #7224

Before this change the preview changes in the buffer viewport were
limited to the size of the preview window ('cmdwinheight').

closes #7220

ckelsel committed:
vim-patch:8.0.0209

Problem:    When using :substitute with the "c" flag and 'cursorbind' is set
            the cursor is not updated in other windows.
Solution:   Call do_check_cursorbind(). (Masanori Misono)

https://github.com/vim/vim/commit/41baa7983aa81b0343b053e6a672cf8224a10245

jamessan committed:
Merge pull request #7227 from ckelsel/vim-8.0.0209

vim-patch:8.0.0209

justinmk committed:
doc: eventloop

justinmk committed:
eventloop: FocusGained: schedule the schedule

main_loop.fast_events does not manifest as K_EVENT, because it is
processed at a different stage than main_loop.events. In order to queue
into main_loop.events, we need to go through the threadsafe
loop_schedule(), which queues into main_loop.thread_events and
eventually main_loop.fast_events. _Then_ it is safe to directly queue
into main_loop.events.

This makes it more likely that the event is treated as K_EVENT.

justinmk committed:
test: use global_helpers.uname() instead of calling to nvim UUT

justinmk committed:
test: FocusGained/FocusLost

justinmk committed:
eventloop: do not redraw in cmdline K_EVENT handler

If :echo is done by an timer or event (such as FocusGained/FocusLost),
redrawcmdline() clobbers it.

justinmk committed:
eventloop: restore redraw in cmdline K_EVENT handler

Restores behavior from commit: 02e86ef04cc1

justinmk committed:
log: introduce `context`

hardenedapple committed:
eventloop: K_EVENT should not finish operator

normal_finish_command() and normal_prepare() assume that any pending
operator needs to be finished after any subsequent key.

Set `finish_op = false` in nv_event() to indicate that the pending
operator shouldn't be finished in normal_execute().

This is how nv_visual() indicates that 'v' or 'V' in operator-pending
mode should not finish the current pending operator.

fixes #5398
fixes #6166 (partially; mappings are still interrupted)

justinmk committed:
lint

justinmk committed:
doc: channel, eventloop

ckelsel committed:
vim-patch:8.0.0247 (#7232)

Problem:    Under some circumstances, one needs to type Ctrl-N or Ctrl-P twice
            to have a menu entry selected. (Lifepillar)
Solution:   call ins_compl_free(). (Christian Brabandt, closes vim/vim#1411)

https://github.com/vim/vim/commit/aed6d0b81a14a81433c0f3c2c65cef935100db33

justinmk committed:
test: FocusGained: let UI settle before sending input

This significantly increases the likelihood that the :echo'd text will
be included in the next UI flush, instead of being lost.

justinmk committed:
eventloop: loop_schedule_deferred()

Generalize the "schedule schedule" technique.

yjhan96 committed:
vim-patch:8.0.0125

Problem:    Not enough testing for entering Ex commands.
Solution:   Add test for CTRL-\ e {expr}. (Dominique Pelle)

https://github.com/vim/vim/commit/eaaa9bbda6ec0a8589a9b23720f95bffe01dc267

jamessan committed:
Merge pull request #7233 from yjhan96/vim-8.0.0125

vim-patch:8.0.0125

justinmk committed:
test: lint

msva committed:
tui: DECSCUSR workaround for Konsole (#7236)

closes #7235

justinmk committed:
eventloop: K_EVENT does not finish mapping

The "mapping" tests added in 541dde36e330 were flawed:
- Unlike op-pending mode, RPCs are _blocked_ during map-pending. So
  a synchronous RPC like nvim_get_current_buf() waits until
  'timeoutlen', then the mapping is canceled.
- helpers.expect() also performs a blocking RPC, so again that must not
  intervene the two nvim_input() calls.

closes #6166

justinmk committed:
Merge #5014 ':terminal : handle F1-F12, other keys'

closes #3101
closes #4343
closes #5024
closes #5925

justinmk committed:
test: FocusGained: press-enter prompt

ckelsel committed:
vim-patch: 8.0.0{181,182,188} (#7152)

vim-patch:8.0.0181
Problem:    When 'cursorbind' and 'cursorcolumn' are both on, the column
            highlignt in non-current windows is wrong.
Solution:   Add validate_cursor(). (Masanori Misono, closes vim/vim#1372)
https://github.com/vim/vim/commit/519d7785f4437762c07b2e04217f83a069a8c663

vim-patch:8.0.0182
Problem:    When 'cursorbind' and 'cursorline' are set, but 'cursorcolumn' is
            not, then the cursor line highlighting is not updated. (Hirohito
            Higashi)
Solution:   Call redraw_later() with NOT_VALID.
https://github.com/vim/vim/commit/e47683a0913f102b6ae08c8848d5aa675d99b188

vim-patch:8.0.0188
    Problem:    Using NOT_VALID for redraw_later() to update the cursor
                line/column highlighting is not efficient.
    Solution:   Call validate_cursor() when 'cul' or 'cuc' is set.
https://github.com/vim/vim/commit/9506cad7a1a6e52a613f356de969cbd980815777

justinmk committed:
Merge #7221 from justinmk/ev-focusgained

tui: schedule event instead of <FocusGained> pseudokey

justinmk committed:
Merge #7241 ':terminal : handle F1-F12, other keys'

justinmk committed:
test: ex_terminal_spec.lua: retry flaky test (#7245)

https://api.travis-ci.org/jobs/271833660/log.txt

KillTheMule committed:
'inccommand': fix 'gdefault' lockup #7261

closes #7244
ref #7249

justinmk committed:
runtime/syntax/vim.vim: disable g:vimsyn_embed by default

This feature is extremely slow.
Also merge some minor upstream differences (missed by a previous merge).

justinmk committed:
runtime/syntax/vim.vim: highlight vimEmbedError as Normal

justinmk committed:
terminal.c: need maketitle() in terminal-mode

normal_redraw() usually takes care of this, but that doesn't happen
during terminal-mode.

regression by c484323dc67f
steps to reproduce:
    nvim -u NORC --cmd 'execute("set titlestring=" . $NVIM_LISTEN_ADDRESS) | set title | startinsert | !sleep 1' term://sh

closes #7248

justinmk committed:
ci/travis: report python environment info

justinmk committed:
test: FocusGained: retry() cmdline-mode test

justinmk committed:
ci/travis: skip pip3 upgrade if pip3 is missing

Workaround for travis issue:
https://github.com/travis-ci/travis-ci/issues/8363

Sometimes `pip3` works, sometimes not:
   pyenv: pip3: command not found
   The `pip3' command exists in these Python versions:
     3.5
     3.5.3

Tried these steps to fix the issue:
- add `python: 3.6` to top level of `.travis.yml`
- add `python3` to `addons.apt.packages` level of `.travis.yml`
- `pyenv global system 3.{4,5,6}`
- `pyenv global 3.6`
In all cases the presence or absence of `pip3` was random.

justinmk committed:
Merge #7253 from justinmk/ci-travis

ckelsel committed:
vim-patch:8.0.0282 vim-patch:8.0.0291 (#7255)

vim-patch:8.0.0291

Problem:    Visual block insertion does not insert in all lines.
Solution:   Don't bail out of insert too early. Add a test. (Christian
            Brabandt, closes vim/vim#1290)

https://github.com/vim/vim/commit/23fa81d2223cd9bb7c51829c48047b2976bc2d11

vim-patch:8.0.0282

Problem:    When doing a Visual selection and using "I" to go to insert mode,
            CTRL-O needs to be used twice to go to Normal mode.
            (Coacher)
Solution:   Check for the return value of edit(). (Christian Brabandt,
            closes #1290)

https://github.com/vim/vim/commit/0b5c93a7f266cd8c90ea27bdaf9f7214a95d64d7

ckelsel committed:
vim-patch:8.0.0186 (#7154)

Problem:    The error message from assert_notequal() is confusing.
Solution:   Only mention the expected value.

https://github.com/vim/vim/commit/5869cf060e60cc09e71b2b3bd85f0576ec78f9f5

justinmk committed:
ci/travis: ignore pip3 failure

Workaround for travis issue:
https://github.com/travis-ci/travis-ci/issues/8363

Cannot check `command -v pip3`, because that may point to
`/opt/pyenv/shims/pip3` which is also (sometimes) broken.

ref 6389bde0bc96

ckelsel committed:
Vim-patch 8.0.0300 (#7258)

vim-patch:8.0.0300

Problem:    Cannot stop diffing hidden buffers. (Daniel Hahler)
Solution:   When using :diffoff! make the whole list if diffed buffers empty.
            (closes vim/vim#736)

https://github.com/vim/vim/commit/25ea0544587dd45088512fec9d7a685e35fea679

justinmk committed:
ex_cmds.c:do_sub(): macroize duplicate code

justinmk committed:
Merge #7262 'inccommand': fix 'gdefault' lockup

mqudsi committed:
build: show a hint for BSD make (#7275)

BSD Make will give preference to a BSDmakefile in the same directory
over a generic Makefile; this can be used to instruct BSD users to build
neovim with GNU Make (gmake) instead.

Otherwise, a flood of syntax errors stemming from the GNU-specific
Makefile will be displayed - which most BSD users are accustomed to, but
may confuse beginners nevertheless.

justinmk committed:
getchar.c: add TERM_FOCUS to MAP_HASH (#7271)

vim-patch:8.0.1108

https://github.com/vim/vim/commit/69fbc9e1dab176f345719436cd89d854df0a2abd

justinmk committed:
test: more coverage for RPC + op-pending #3732

jamessan committed:
tui: Use unibi_var_from_num when available

As of unibilium 1.2.1, directly manipulating unibi_var_t is deprecated.

    ../src/nvim/tui/tui.c: In function 'update_attrs':
    ../src/nvim/tui/tui.c:321:7: warning: 'i' is deprecated: use unibi_var_from_num or unibi_num_from_var instead [-Wdeprecated-declarations]
           data->params[0].i = (fg >> 16) & 0xff;  // red
           ^~~~
    In file included from ../src/nvim/tui/tui.c:12:0:
    /usr/include/unibilium.h:632:9: note: declared here
         int i   UNIBI_DEPRECATED("use unibi_var_from_num or unibi_num_from_var instead");
             ^

All use should go through unibi_{num,str}_from_var and
unibi_var_from_{num,str}.  Wrap access of unibi_var_t behind a new
UNIBI_SET_NUM_VAR macro which uses the new functions when they're
available.

bfredl committed:
Revert cbda7d8 "build: Revert 464bc16."

In-tree builds are no longer allowed, so deleting build/docs/ is
harmless

ref 53eddb8

ckelsel committed:
vim-patch:8.0.0305 (#7265)

Problem:    Invalid memory access when option has duplicate flag.
Solution:   Correct pointer computation. (Dominique Pelle, closes vim/vim#1442)

https://github.com/vim/vim/commit/aaaf57d8a936efe420190c077e4a74041cc6c72e

ckelsel committed:
vim-patch:8.0.0302 NA (#7264)

(NA: Nvim removed the ability to set termcodes directly.)

Problem:    Cannot set terminal key codes with :let.
Solution:   Make it work.

https://github.com/vim/vim/commit/e353c402e63b9b0a0bc06acf390e352d9e7eeaeb

ckelsel committed:
vim-patch:8.0.0324 (#7279)

Problem:    Illegal memory access with "1;y".
Solution:   Call check_cursor() instead of check_cursor_lnum(). (Dominique
            Pelle, closes vim/vim#1455)

https://github.com/vim/vim/commit/f1f6f3f7df2938b3583e341482d96c1d53124c51

ckelsel committed:
vim-patch:8.0.0327 (#7281)

Problem:    The E11 error message in the command line window is not
            translated.
Solution:   use _(). (Hirohito Higashi)

https://github.com/vim/vim/commit/75c19464ed7fb6024af64747379e61abc4e4a483

jamessan committed:
Merge pull request #7277 from jamessan/deprecated-unibilium-APIs

tui: Use unibi_var_from_num when available

bfredl committed:
Merge pull request #7280 from bfredl/fix_helptags

cmake: remove stale doc files to avoid "duplicate tags" message

justinmk committed:
Merge #7252 from justinmk/perf-vim.vim

euclio committed:
escape amatch filename (#7292)

Fixes #7046
Fixes autozimu/LanguageClient-neovim#77

jamessan committed:
cmake: Check if the compiler understands -Wno-array-bounds

Closes #7297

blueyed committed:
doc: fix typo with :lcd (#7299)

jamessan committed:
Move definition of cstrchr above the functions which call it

Functions with FUNC_ATTR_ALWAYS_INLINE need to be defined before they
are called to work around bugs with some compiler versions.  When the
body is after the use of the function, compilation will fail with

    /home/niko/build/neovim/src/nvim/regexp.c: In function 'regmatch':
    /home/niko/build/neovim/build/src/nvim/auto/regexp.c.generated.h:77: sorry, unimplemented: inlining failed in call to 'cstrchr': function body not available
    /home/niko/build/neovim/src/nvim/regexp.c:4193: sorry, unimplemented: called from here

tjrana committed:
runtime/tutor: fix typos (#7302)

DarkDeepBlue committed:
vim-patch:8.0.0294 (#7305)

Problem:    Argument list is not stored correctly in a session file.
            (lgpasquale)
Solution:   Use "$argadd" instead of "argadd". (closes vim/vim#1434)

https://github.com/vim/vim/commit/79da563cf9220b9abb83455a68d995684133ea56
Signed-off-by: Michael Schupikov <michael@schupikov.de>

jamessan committed:
Merge pull request #7298 from jamessan/check-array-bounds-support

cmake: Check if the compiler understands -Wno-array-bounds

ckelsel committed:
vim-patch:8.0.0328

Problem:    The "zero count" error doesn't have a number. (Hirohito Higashi)
Solution:   Give it a number and be more specific about the error.

https://github.com/vim/vim/commit/23a5558cfd860401aa694f0302d621887440f031

ckelsel committed:
vim-patch:8.0.0330

Problem:    Illegal memory access after "vapo". (Dominique Pelle)
Solution:   Fix the cursor column.

https://github.com/vim/vim/commit/84b2a381451e9068b09ef6d85f5e8cf1598e7355

ckelsel committed:
vim-patch:8.0.0333

Problem:    Illegal memory access when 'complete' ends in a backslash.
Solution:   Check for trailing backslash. (Dominique Pelle, closes vim/vim#1478)

https://github.com/vim/vim/commit/226c53429109f24e31c17016aedfd7fbf7a9aa50

ckelsel committed:
vim-patch:8.0.0091 (#7312)

Problem:    Test_help_complete sometimes fails in MS-Windows console.
Solution:   Use getcompletion() instead of feedkeys() and command line
            completion. (Hirohito Higashi)

https://github.com/vim/vim/commit/9f0e423c2818c0cacd0810f9c3c67cbb6b80963d

KillTheMule committed:
test: lint whitespace in legacy/ (#7308)

jamessan committed:
Merge pull request #7311 from ckelsel/vim-8.0.0333

vim-patch:8.0.0333

jamessan committed:
vim-patch:8.0.0112

Problem:    Tests 92 and 93 are old style.
Solution:   Make test92 and test93 new style. (Hirohito Higashi, closes vim/vim#1289)

https://github.com/vim/vim/commit/eca626fcdb73d480660c78b9f84cc043fa561922

jamessan committed:
vim-patch:8.0.0253

Problem:    When creating a session when winminheight is 2 or larger and
            loading that session gives an error.
Solution:   Also set winminheight before setting winheight to 1. (Rafael
            Bodill, neovim vim/vim#5717)

https://github.com/vim/vim/commit/36ae89c550a4f0a380606c3fb4a054957ad698f0

jamessan committed:
vim-patch:8.0.0258

Problem:    mksession test leaves file behind.
Solution:   Delete the file.  Rename files to start with "X".

https://github.com/vim/vim/commit/c9b56b2ceb4662f87c39ea07ba5090a073fd6286

jamessan committed:
vim-patch:8.0.1024

Problem:    Manual folds are lost when a session file has the same buffer in
            two windows. (Jeansen)
Solution:   Use ":edit" only once. (Christian Brabandt, closes vim/vim#1958)

https://github.com/vim/vim/commit/4bebc9a0565670b853d227f81a9a31eafdb47eed

jamessan committed:
vim-patch:8.0.1025

Problem:    Stray copy command in test.
Solution:   Remove the copy command.

https://github.com/vim/vim/commit/4e83961985abb78757b135f29ac4ffde675247af

jamessan committed:
lint

jamessan committed:
Merge pull request #7309 from ckelsel/vim-8.0.0328

vim-patch:8.0.0328

jamessan committed:
Merge pull request #7310 from ckelsel/vim-8.0.0330

vim-patch:8.0.0330

jamessan committed:
Merge pull request #7325 from jamessan/vim-8.0.0112

vim-patch:8.0.0112,8.0.0253,8.0.0258,8.0.1024,8.0.1025

ckelsel committed:
vim-patch:8.0.0131

Problem:    Not enough test coverage for syntax commands.
Solution:   Add more tests. (Dominique Pelle)

https://github.com/vim/vim/commit/73b484c4da00011317dc68ada4f5dfc6515ad263

ckelsel committed:
vim-patch:8.0.0132

Problem:    Test fails because of using :finish.
Solution:   Change to return.

https://github.com/vim/vim/commit/4c8980b717f73042f1d625ee255fa74eddb989ba

ckelsel committed:
fix conflict
